pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,441 @@
+        self.on_view_mode_change("2D")
+
+    @property
+    def color_map(self):
+        return self.state.cmap
+    
+    @color_map.setter
+    def color_map(self, cmap):
+        self.state.cmap = cmap
+        self.on_colormap_change(cmap)
+
+    @change("cmap")
+    def on_colormap_change(self, cmap, **_):
+        use_preset(self.slice_actor, self.data_actor, cmap)
+        self.ctrl.view_update() 
+
+    @change("logscale")
+    def on_log_scale_change(self, logscale, **_):
+        update_preset(self.slice_actor, logscale)
+        self.ctrl.view_update() 
+
+    @change("view_mode")
+    def on_view_mode_change(self, view_mode, **_):
+        slice_dim = self.state.slice_dim
+        slice_i     = 0 if self.dims[slice_dim] == 'x' else 1 if self.dims[slice_dim] == 'y' else 2
+
+        extents = list(self.extents.values())
+        origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ]
+
+        camera = self.renderer.GetActiveCamera()
+        pos = camera.GetPosition()
+        norm = np.linalg.norm(np.array(origin) - np.array(pos))
+        print(f"Origin and norm : {origin, norm}")
+        if self.state.view_mode == '3D':
+            camera.SetPosition(norm, norm, norm)
+            camera.SetViewUp(0., 1., 0.)
+        elif self.state.view_mode == '2D':
+            self.renderer.RemoveActor(self.outline_actor)
+            self.renderer.RemoveActor(self.data_actor)
+            if slice_i == 0:
+                camera.SetPosition(norm, origin[1], origin[2])
+                camera.SetViewUp(0., 1., 0.)
+            elif slice_i == 1:
+                camera.SetPosition(origin[0], norm, origin[2])
+                camera.SetViewUp(0., 0., 1.)
+            elif slice_i == 2:
+                camera.SetPosition(origin[0], origin[1], norm)
+                camera.SetViewUp(0., 1., 0.)
+            camera.SetFocalPoint(origin[0], origin[1], origin[2])
+            camera.OrthogonalizeViewUp()
+        self.renderer.SetActiveCamera(camera)
+        self.renderer.ResetCamera()
+        self.ctrl.view_update()
+
+    @change("data_var", "time_active", "dimval")
+    def on_data_change(self, **_):
+        dimval    = self.state.dimval
+        slice_dim = self.state.slice_dim
+        normal      = [0, 0, 0]
+        slice_i     = 0 if self.dims[slice_dim] == 'x' else 1 if self.dims[slice_dim] == 'y' else 2
+        normal[slice_i] = 1
+        
+        extents = list(self.extents.values())
+        origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ] 
+        origin[slice_i] = dimval
+        
+        self.plane.SetOrigin(origin)
```

Comment:
do we need to update origin/normal?

Reply:
The plane's origin, and normal is what the cutter uses to get the slice. I think every time the user changes the slice value along a dimension, we need to update the `origin`, and every time the dimension is updated, the `normal` needs to change as well.